# Single cell papers with code

Single cell papers with code can not only facilitate the reproducibility of biomedical researches, but also promote our skills of analyzing single cell data. 


'Papers with code' here means that authors provide necessary codes to reproduce figures or results in their papers.

(Last update: Jan. 6th, 2021)

### Tutorial

- [[code | Python]](https://github.com/theislab/single-cell-tutorial) Luecken, M.D., and Theis, F.J. (2019). **Current best practices in single‐cell RNA‐seq analysis: a tutorial.** Mol. Syst. Biol. 15, e8746.

### Benchmark study

- [[code | R]](https://github.com/oxwang/fda_scRNA-seq) Chen, W. et al. [**A multicenter study benchmarking single-cell RNA sequencing technologies using reference samples.**](https://www.nature.com/articles/s41587-020-00748-9) Nat. Biotechnol. (2020) doi:10.1038/s41587-020-00748-9.

### Trajectory analysis

- [[code | Python]](https://github.com/rajewsky-lab/planarian_lineages) Plass, M., Solana, J., Alexander Wolf, F., Ayoub, S., Misios, A., Glažar, P., Obermayer, B., Theis, F.J., Kocks, C., and Rajewsky, N. (2018). **Cell type atlas and lineage tree of a whole complex animal by single-cell transcriptomics.** Science (80-. ). 360.

### Cell Atlas

- [[code | R]](https://github.com/Single-Cell-Genomics-Group-CNAG-CRG/Tumor-Immune-Cell-Atlas) Nieto, P., Marc Elosua-Bayes, M., Trincado, J.L., Marchese, D., Massoni-Badosa, R., Salvany, M., Henriques, A., Mereu, E., Moutinho, C., Ruiz, S., et al. (2020). **A Single-Cell Tumor Immune Atlas for Precision Oncology.** BioRxiv 2020.10.26.354829.

- [[code | Python]](https://github.com/czbiohub/tabula-muris-senis) Tabula Muris Consortium. **A single-cell transcriptomic atlas characterizes ageing tissues in the mouse.** Nature 2020;583. https://doi.org/10.1038/s41586-020-2496-1.

- [[code | R]](https://github.com/krasnowlab/HLCA) Travaglini, K. J. et al. **A molecular cell atlas of the human lung from single-cell RNA sequencing.** Nature 587, 619–625 (2020).


- [[code | Python and R]](https://github.com/haniffalab/HCA_skin) Reynolds, G. et al. [**Poised cell circuits in human skin are activated in disease.**](https://www.biorxiv.org/content/10.1101/2020.11.05.369363v1) bioRxiv 2020.11.05.369363 (2020) doi:10.1101/2020.11.05.369363.


### Developmental biology

- [[code | R]](https://github.com/ZornLab/Single-cell-transcriptomics-reveals-a-signaling-roadmap-coordinating-endoderm-and-mesoderm-lineage) Han, L., Chaturvedi, P., Kishimoto, K., Koike, H., Nasr, T., Iwasawa, K., Giesbrecht, K., Witcher, P.C., Eicher, A., Haines, L., et al. (2020). **Single cell transcriptomics identifies a signaling network coordinating endoderm and mesoderm diversification during foregut organogenesis.** Nat. Commun. 11.

### Neuroscience

- [[code | Matlab]](https://github.com/linnarsson-lab/DG_paper_public) Hochgerner, H., Zeisel, A., Lönnerberg, P., and Linnarsson, S. (2018). **Conserved properties of dentate gyrus neurogenesis across postnatal development revealed by single-cell RNA sequencing.** Nat. Neurosci. 21.

- [[code | R]](https://github.com/alexandrotrevino/brainchromatin) Trevino, A. E. et al. [**Chromatin and gene-regulatory dynamics of the developing human cerebral cortex at single-cell resolution.**](https://www.biorxiv.org/content/10.1101/2020.12.29.424636v2) bioRxiv 2020.12.29.424636 (2021) doi:10.1101/2020.12.29.424636.


### Experimental research

- [[code | R]](https://github.com/tgen/banovichlab) Habermann A, Gutierrez A, Bui L, Yahn S, Winters N, Calvi C, et al. **Single-cell RNA sequencing reveals profibrotic roles of distinct epithelial and mesenchymal lineages in pulmonary fibrosis.** Sci Adv 2020;6:eaba1972. https://doi.org/10.1101/753806.


- [[code | R]](https://github.com/marzamKI/neurogenic_astros) Zamboni, M., Llorens-Bobadilla, E., Magnusson, J.P., and Frisén, J. (2020). **A Widespread Neurogenic Potential of Neocortical Astrocytes Is Induced by Injury.** Cell Stem Cell 1–13.


- [[code | Python]](https://github.com/linnarsson-lab/ipynb-lamanno2016) La Manno, G., Gyllborg, D., Codeluppi, S., Nishimura, K., Salto, C., Zeisel, A., Borm, L.E., Stott, S.R.W., Toledo, E.M., Villaescusa, J.C., et al. (2016). **Molecular Diversity of Midbrain Development in Mouse, Human, and Stem Cells.** Cell 167, 566-580.e19.

- [[code | R]](https://github.com/klarman-cell-observatory/ivPerturbSeq) Jin, X. et al. **In vivo Perturb-Seq reveals neuronal and glial abnormalities associated with autism risk genes.** Science (80-. ). 370, eaaz6063 (2020).

### Spatial transcriptomics

- [[code | R]](https://github.com/saezlab/visium_heart/tree/master/snRNA_seq) Kuppe, C. et al. **Spatial multi-omic map of human myocardial infarction.** bioRxiv 2020.12.08.411686 (2020) doi:10.1101/2020.12.08.411686.


### Visualization

- [[code]](https://github.com/faryabib/NatMethods_TooManyCells_analysis) Schwartz GW, Zhou Y, Petrovic J, Fasolino M, Xu L, Shaffer SM, et al. **TooManyCells identifies and visualizes relationships of single-cell clades.** Nat Methods 2020. https://doi.org/10.1038/s41592-020-0748-5.


